saleor/core/tests/test_taxes.py

```diff
@@ -0,0 +1,270 @@
+from unittest.mock import Mock, patch
+
+import pytest
+
+from saleor.app.models import App
+from saleor.core.taxes import (
+    DEFAULT_TAX_CODE,
+    DEFAULT_TAX_DESCRIPTION,
+    WEBHOOK_TAX_CODES_CACHE_KEY,
+    TaxType,
+    _get_cached_tax_codes_or_fetch,
+    _get_current_tax_app,
+    get_tax_type,
+    set_tax_code,
+)
+from saleor.webhook.event_types import WebhookEventAsyncType, WebhookEventSyncType
+from saleor.webhook.models import Webhook, WebhookEvent
+
+
+@pytest.fixture
+def app_factory():
+    def factory(name, is_active, webhook_event_types):
+        app = App.objects.create(name=name, is_active=is_active)
+        webhook = Webhook.objects.create(
+            name=f"{name} Webhook",
+            app=app,
+            target_url="https://test.webhook.url",
+        )
+        for event_type in webhook_event_types:
+            WebhookEvent.objects.create(
+                webhook=webhook,
+                event_type=event_type,
+            )
+        return app
+
+    return factory
+
+
+@pytest.fixture
+def tax_app_factory(app_factory):
+    def factory(name, is_active):
+        return app_factory(
+            name=name,
+            is_active=is_active,
+            webhook_event_types=[
+                WebhookEventSyncType.CHECKOUT_CALCULATE_TAXES,
+                WebhookEventSyncType.ORDER_CALCULATE_TAXES,
+            ],
+        )
+
+    return factory
+
+
+@pytest.fixture
+def tax_app(tax_app_factory):
+    return tax_app_factory(
+        name="Tax App",
+        is_active=True,
+    )
+
+
+def test_get_current_tax_app(tax_app):
+    # given
+    expected_app = tax_app
+
+    # when
+    app = _get_current_tax_app()
+
+    # then
+    assert expected_app == app
+
+
+def test_get_current_tax_app_multiple_apps(app_factory, tax_app_factory):
+    # given
+    tax_app_factory(
+        name="Another Tax App",
+        is_active=True,
+    )
+    expected_app = tax_app_factory(
+        name="Tax App",
+        is_active=True,
+    )
+    app_factory(
+        name="Non Tax App",
+        is_active=True,
+        webhook_event_types=[
+            WebhookEventAsyncType.ORDER_UPDATED,
+        ],
+    )
+    app_factory(
+        name="Partial Tax App",
+        is_active=True,
+        webhook_event_types=[
+            WebhookEventSyncType.CHECKOUT_CALCULATE_TAXES,
+        ],
+    )
+    tax_app_factory(
+        name="Inactive Tax App",
+        is_active=False,
+    )
+
+    # when
+    app = _get_current_tax_app()
+
+    # then
+    assert expected_app == app
+
+
+def test_get_current_tax_app_no_app():
+    # when
+    app = _get_current_tax_app()
+
+    # then
+    assert app is None
```

Comment:
Like mention above it could be one liner.

Reply:
Resolved in recent commit.